js/common/tests/shell/shell-document.js

```diff
@@ -2338,6 +2338,44 @@ function DatabaseDocumentSuiteReturnStuff () {
       assertTypeOf("string", res2._rev);
     },
 
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief use overwrite option
+////////////////////////////////////////////////////////////////////////////////
+
+    testInsertOverwrite : function () {
+      var docHandle = collection.insert({ a : 1});
+      var key = docHandle._key;
+
+      // normal insert with same key must fail!
+      try{
+        var res = collection.insert({a : 2, _key : key});
+        fail();
+      }
+      catch (err) {
+        assertEqual(ERRORS.ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED.code, err.errorNum);
+      }
+
+      // overwrite with same key must work
+      collection.insert({a : 2, _key: key},{overwrite:true});
+      var arr = collection.toArray();
+      assertEqual(arr.length, 1);
+      assertEqual(arr[0].a, 2);
+
+      // overwrite (babies) with same key must work
+      collection.insert({a : 2, _key: key},{overwrite:true});
+      arr = collection.toArray();
+      assertEqual(arr.length, 1);
+      assertEqual(arr[0].a, 2);
+
+      // overwrite (babies) with same key must work
+      collection.insert([{a : 3, _key: key}, {a : 4, _key: key}, {a : 5, _key: key}], {overwrite:true});
```

Comment:
same here: we should test that it's actually replace, not an update!

Reply:
additionally a test is missing for a repsert (overwrite: true) for which the document does not initially exist.

Reply:
should we add a test for a repsert (overwrite: true) when the inserted document does not contain a `_key` attirubte?